### PR TITLE
itest: extend multi-input PSBT spend test to include partial value spend

### PR DESCRIPTION
**Pull request by ffranr**
This commit extends the existing multi-input PSBT spend test to also include a partial value spend.

### CODE REVIEW

File: itest/psbt_test.go [hunk 1]
```diff
@@ -1022,13 +1032,38 @@ func testMultiInputPsbtSingleAssetID(t *harnessTest) {
...
 	AssertNonInteractiveRecvComplete(t.t, secondaryTapd, 2)
 
-	t.Logf("Two separate send events complete, now attempting to send " +
-		"back the full amount in a single multi-input PSBT send event")
+	// Third of three send events from primary (minting) node to the
+	// secondary node.
+	sendAmt = uint64(3000)
+	changeAmt = uint64(0)
+
+	addr, err = secondaryTapd.NewAddr(
```
Comment (guggero) [inline itest/psbt_test.go:1045]: If the goal is to just create three different on-chain outputs, we _could_ use a multi-send (`sendAssetsToAddr()` supports multiple receiver addresses). Would shorten the code a bit.

**Review by guggero — APPROVED** [review]
Very nice! LGTM :tada:

File: itest/psbt_test.go (hunk 1 above)
Comment (ffranr) [reply itest/psbt_test.go:1045]: Using `sendAssetsToAddr` I ran into problems reusing an address. (Not all output script keys were unique.) I'll investigate and try to get it working but I'll leave this PR as is for now.

Comment (jharveyb) [reply itest/psbt_test.go:1045]: Why would address reuse be necessary? IIUC each call to `NewAddr` should be returning addresses with unique script keys.

File: itest/psbt_test.go [hunk 2]
```diff
@@ -1108,6 +1150,82 @@ func testMultiInputPsbtSingleAssetID(t *harnessTest) {
...
+	t.Logf("Partial amount multi-input PSBT send event complete. Now " +
+		"attempting to send the remaining amount in a full value " +
+		"multi-input PSBT send event back to the primary node from " +
+		"the secondary node.")
+
+	// Attempt a full value send of the rest of the asset back to the
+	// primary node.
+	sendAmt = uint64(1500)
+
+	vPkt = tappsbt.ForInteractiveSend(
```
Comment (jharveyb) [inline itest/psbt_test.go:1163]: super-nit: we could assert the # of inputs on these PSBTs;  IIUC should be two in both cases?

**Review by jharveyb — APPROVED** [review]
LGTM, solid extension of coverage 💯

File: itest/psbt_test.go [hunk 3]
```diff
@@ -1001,17 +1007,21 @@ func testMultiInputPsbtSingleAssetID(t *harnessTest) {
...
+		genInfo.AssetId, []uint64{changeAmt, sendAmt}, 0, 1,
 	)
 
 	AssertNonInteractiveRecvComplete(t.t, secondaryTapd, 1)
 
-	// Second of two send events from primary (minting) node to the
-	// secondary node.
+	// Second of three send events from primary (minting) node to secondary
+	// node.
+	sendAmt = uint64(1000)
```
Comment (GeorgeTsagk) [inline itest/psbt_test.go:1017]: nit: instead of hard-coding the numbers here maybe calculate them from previous `changeAmt`?

**Review by GeorgeTsagk — APPROVED** [review]
Lgtm!

File: itest/psbt_test.go (hunk 2 above)
Comment (ffranr) [reply itest/psbt_test.go:1163]: good idea, added

File: itest/psbt_test.go (hunk 3 above)
Comment (ffranr) [reply itest/psbt_test.go:1017]: I've added a running total for more dynamic change amount calculation.

File: itest/psbt_test.go (hunk 1 above)
Comment (ffranr) [reply itest/psbt_test.go:1045]: I might try again without address re-use. Thanks for pointing out the unique script key property. I'll look into that.

**Pull request merged**